### ISSUE TITLE
Lingering socket connection prevents app from exiting

### ISSUE THREAD

**Issue by pmidge** [Instrumental/instrumental_agent-node#16]
Just switched from the old client to `instrumental_agent-node` and I've noticed that my app no longer completely exits. I believe this is due to a ref'd socket in the Instrumental module.

In my case the underlying Instrumental object will never be GC'd since it's rooted to an object that only goes out of scope on exit. I've added a call in that code's shutdown routine (this is a long-running process that has graceful shutdown behavior) to invoke `Instrumental.socket.end()`, solving the problem.

However, this gives me hives since I feel like I'm accessing a private class member, even though this is JS and nobody seems to care much about that. Still... ew.

I'm happy to take a stab at solving this in the module and make a PR but would want your input before starting:

1. Setting a socket timeout and setting `this.socket = undefined` after destroying the socket. The API already has robust reconnection logic so this would be transparent to the user. There is potential impact to the Instrumental service getting connection flooded by clients with ridiculously low timeouts, but enforcing a reasonable range should get around this.

2. Hooking `SIGINT` and closing the connection.

3. Exposing something like `Instrumental.disconnect()` and leave it up to the caller when to use it.

...or maybe something else. I'm happy to keep using my workaround, but I imagine others will hit this.

**Comment by janxious** [comment]
Sorry that is happening. We do our best not to break people's worlds in any way. 😐

PR would be appreciated, and hopefully you can test the branch change out to make sure it all works as expected in your env. As to your suggestions, it seems like creating something like `Instrumental.disconnect()`, and then using it in the `SIGINT` and in a timeout would be the best move. So basically all 3. 

As to the fear of connection flooding, I'm not super concerned about it. If it becomes an issue we have monitoring on the situation and the ability to add scale real quick and in a hurry.

**Comment by janxious** [comment]
Fixed by #17
